### PR TITLE
Audio for Videos

### LINKED ISSUE #1489: Audio for Videos
Audio-compatible video formats should play audio and have a mute toggle.

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1489 

# Pull Request Information

Enhances video handling:
- mp4s can now play audio
- Videos with audio now show a mute button on the top right
- Default mute behavior can be controlled in settings (General -> Mute Videos)
- Tapping videos now pauses them instead of stops them
- mp4s now loop

## Known Issues
- Interactive dismissal causes the video to start over and stop almost immediately. If audio is on, this causes a short clip of audio to play. This is due to an issue with our sheet dismissal: #1492

### CODE REVIEW

File: Mlem/App/Views/Shared/Images/Helpers/AnimationControlLayer.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            .overlay(alignment: .topTrailing) {
+                if let muted {
+                    Image(systemName: muted.wrappedValue ? Icons.muted : Icons.unmuted)
+                        .resizable()
+                        .scaledToFit()
+                        .frame(width: 15, height: 15)
+                        .padding(5)
+                        .background {
+                            Circle().foregroundStyle(.ultraThinMaterial)
+                        }
```
Comment (Sjmarf) [inline Mlem/App/Views/Shared/Images/Helpers/AnimationControlLayer.swift]: *nit* this is slightly cleaner imo

```suggestion
                        .background(.ultraThinMaterial, in: .circle)
```

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Nice work! 

Pressing the volume up button whilst playing a muted video unmutes the `AVQueuePlayer`, but the mute button in the UI doesn't get updated. Maybe we need to watch `player.muted` for changes and update `muted` if so? Could be tricky though if a view update doesn't get triggered when `player.muted` changes.

**Comment by EricBAndrews** [comment]
Ooh, nice catch! I've added some observation code that should sort it out.

**Review by Sjmarf — APPROVED** [review]
LGTM! It would be nice if videos were paused once they leave the screen. If I unmute a video and scroll away, the audio keeps playing.

**Pull request merged**